base_exception/models/base_exception.py

```diff
@@ -162,16 +163,16 @@ def detect_exceptions(self):
     def _exception_rule_eval_context(self, obj_name, rec):
         user = self.env['res.users'].browse(self._uid)
         return {obj_name: rec,
-                'self': self.pool.get(rec._name),
+                'self': self.env.get(rec._name),
                 'object': rec,
                 'obj': rec,
-                'pool': self.pool,
-                'cr': self._cr,
-                'uid': self._uid,
+                'env': self.env,
+                'cr': self.env.cr,
+                'uid': self.env.uid,
```

Comment:
Are you keeping self.env.cr and self.env.uid for backward compatibility? Could it be deprecated to use only env ? Maybe add a note for version 12.0 if we can simplify it.

Or is it to have helpers in `safe_eval`

Reply:
DONE

Reply:
ping @yvaucher

Reply:
Ok self.env.user is now used. Thanks.